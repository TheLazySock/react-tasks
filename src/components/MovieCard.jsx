import React from 'react';
import { Link } from "react-router-dom";

const MovieCard = (props) => {
    const releaseDateYear = new Date(props.release_date).getFullYear();

    const genresString = props.genres.join(' & ');

    return (
        <Link className="movie-card" to={`/movie/${props.id}`}>
            <img className="movie-card-poster" src={props.poster_path} alt="There must be poster"/>
            <div>
                <label className="movie-card-title">{props.title}</label>
                <label className="movie-card-date">{releaseDateYear}</label>
            </div>
            <label className="movie-card-genres">{genresString}</label>
        </Link>
    )
}

export default MovieCard;