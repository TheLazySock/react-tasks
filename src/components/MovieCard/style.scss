.movie-card {
    margin: 25px 40px;
    min-height: 480px;
    height: 100%;
    width: 300px;
    text-decoration: none;
    cursor: default;
    .movie-card-poster {
        // height: 450px;
        width: 100%;
        cursor: pointer;
    }
    & > div {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        color: #000000;
    }
    .movie-card-date {
        padding: 1px 3px;
        border: solid 1px #aeaeae;
        border-radius: 5px;
        width: 32px;
        height: 21px;
    }
    .movie-card-title {
        font-size: 18px;
    }
    .movie-card-genres {
        font-size: 12px;
        color: #7c7c7c;
    }
}