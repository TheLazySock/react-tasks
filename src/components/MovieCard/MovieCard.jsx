// @flow
import React from 'react';
import { Link } from 'react-router-dom';
import CSSModules from 'react-css-modules';

import style from './style.scss';

type MovieCardProps = {
    release_date: Date,
    genres: Array<string>,
    id: number | string,
    poster_path: ?string,
    title: string
}

const MovieCard = (props: MovieCardProps) => {
    const releaseDateYear = new Date(props.release_date).getFullYear();

    const genresString = Array.isArray(props.genres) ? props.genres.join(' & ') : 'No any genres';

    return (
        <Link styleName="movie-card" to={`/movie/${props.id}`}>
            <img styleName="movie-card-poster" src={props.poster_path} alt="There must be poster"/>
            <div>
                <label styleName="movie-card-title">{props.title}</label>
                <label styleName="movie-card-date">{releaseDateYear}</label>
            </div>
            <label styleName="movie-card-genres">{genresString}</label>
        </Link>
    );
};

export default CSSModules(MovieCard, style);
