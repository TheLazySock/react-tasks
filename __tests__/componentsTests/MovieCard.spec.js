import React from 'react';
import renderer from 'react-test-renderer';
import MovieCard from '../../src/components/MovieCard';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('>>>MOVIE CARD --- snapshot', () => {
    const movieInfo = {
        id: 680,
        poster_path: "https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg",
        title: "Pulp Fiction",
        release_date: "1994-09-10",
        genres: ["Thriller", "Crime"]
    }

    it('MovieCard have to rendered succesfully', () => {
        const movieCard = shallow(<MovieCard 
            id={movieInfo.id} 
            poster_path={movieInfo.poster_path}
            title={movieInfo.title}
            release_date={movieInfo.release_date}
            genres={movieInfo.genres}
        />);
        expect(shallowToJson(movieCard)).toMatchSnapshot();
    });

    it('MovieCard link should work correctly', () => {
        const movieCard = shallow(<MovieCard id={movieInfo.id} />);
        expect(movieCard.prop('to')).toBe('/movie/680');
    })
});