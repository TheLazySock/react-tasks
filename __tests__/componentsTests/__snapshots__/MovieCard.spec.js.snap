// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>MOVIE CARD --- snapshot MovieCard have to rendered succesfully 1`] = `
<Link
  className="movie-card"
  replace={false}
  to="/movie/680"
>
  <img
    alt="There must be poster"
    className="movie-card-poster"
    src="https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg"
  />
  <div>
    <label
      className="movie-card-title"
    >
      Pulp Fiction
    </label>
    <label
      className="movie-card-date"
    >
      1994
    </label>
  </div>
  <label
    className="movie-card-genres"
  >
    Thriller & Crime
  </label>
</Link>
`;
